core/primitives/src/receipt.rs

```diff
@@ -113,6 +115,24 @@ pub struct StateStoredReceiptV0<'a> {
     pub metadata: StateStoredReceiptMetadata,
 }
 
+/// The V1 of StateStoredReceipt.
+#[derive(BorshDeserialize, BorshSerialize, PartialEq, Eq, Debug, ProtocolSchema)]
+pub struct StateStoredReceiptV1<'a> {
+    pub receipt: Cow<'a, Receipt>,
+    pub metadata: StateStoredReceiptMetadata,
+    /// Version of buffer metadata that was updated when the receipt was stored.
+    pub queue_metadata_version: QueueMetadataVersion,
```

Comment:
I don't quite get the need for this field. If we are already versioning `StateStoredReceipt`, each new metadata version can increment StateStoredReceipt version or we can make StateStoredReceiptMetadata a versioned enum itself like Block and BlockHeaders.

Reply:
Alright. Initially I used the version of stored receipt, but then I thought that it could be clearer to mark the version of the metadata explicitly. I'll remove it.